plugins/guests/linux/cap/persist_mount_shared_folder.rb

```diff
@@ -0,0 +1,39 @@
+require "vagrant/util"
+
+require_relative "../../../synced_folders/unix_mount_helpers"
+
+module VagrantPlugins
+  module GuestLinux
+    module Cap
+      class PersistMountSharedFolder
+        extend SyncedFolder::UnixMountHelpers
+
+        def self.persist_mount_shared_folder(machine, fstab_folders, mount_type)
+          export_folders = fstab_folders.map do |name, data|
+            guest_path = Shellwords.escape(data[:guestpath])
+            mount_options, mount_uid, mount_gid  =  mount_options(machine, name, guest_path, data)
+            mount_options = "#{mount_options},nofail"
+            {
+              name: name,
+              mount_point: guest_path,
+              mount_type: mount_type,
+              mount_options: mount_options,
+            }
+          end
+
+          fstab_entry = Vagrant::Util::TemplateRenderer.render('guests/linux/etc_fstab', folders: export_folders)
+          # Replace existing vagrant managed fstab entry
+          machine.communicate.sudo("sed -i '/\#VAGRANT-BEGIN/,/\#VAGRANT-END/d' /etc/fstab")
+          machine.communicate.sudo("echo '#{fstab_entry}' >> /etc/fstab")
+
+          fstab_valid = machine.communicate.test("mount -a", sudo: true)
```

Comment:
I think we should mount each share individually if possible rather than doing a `mount -a`.